src/MySqlConnector/Serialization/MySqlSession.cs

```diff
@@ -84,6 +84,8 @@ public void DoCancel(MySqlCommand commandToCancel, MySqlCommand killCommand)
 				// blocking the other thread for an extended duration.
 				killCommand.CommandTimeout = 3;
 				killCommand.ExecuteNonQuery();
+
+				commandToCancel.IsCanceled = true;
```

Comment:
I started messing around with the states, but if you know how you want to tackle this I have `Allow edits from maintainers` turned on, so go for it.